### DISCUSSION TITLE
Show Resource loading report in profiler.

### DISCUSSION THREAD

**Discussion by TheConceptBoy** [godotengine/godot-proposals#4931] [Editor]
### Describe the project you are working on

A 3D game with lot's of materials

### Describe the problem or limitation you are having in your project

Some material is causing the game to stutter, most likely because that material came into view and had to be loaded on the fly.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Allows user to get a report of which resource (material, scene, node, texture or otherwise) which had to be loaded when inspecting the dip in profiler graph view.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

No pseudo-code. This would be engine level.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No that I know of.

### Is there a reason why this should be core and not an add-on in the asset library?

Seems like resources are loaded on the engine level.

**Comment by Zireael07** [comment]
This is likely not resource loading but shader compiling stutter. Anyway would be nice to know which one stutters.

**Comment by YuriSizov** [comment]
> No pseudo-code. This would be engine level.

We still need an implementation proposal for the engine level, otherwise even if we like the idea, this cannot be worked on. A proposal is not just an idea, it's also a technical detail, and a person willing to work on it (doesn't have to be the OP, but it helps).

A stutter can be a million different things, although as mentioned, in Godot it's likely shaders compiling the first time they are executed. I also had a stutter when I used a very large image for the first time. The resources are already loaded at this point, so it's not about that.

Which is where a technical detail would help us discuss it further, so we know what we are trying to track.
